ADAH-EviDENce/evidence
# 44-progress bar for model generation

Refs: #44 
This PR adds progress bars to Jupyter notebooks. The progress is shown both on the terminal, when using the Docker image, and also when running the notebooks interactively.

### To test:
- Build the Docker image
```shell
docker build -t evidence ./model-generation
```
-  Test non-interactively
```shell
export EXPERIMENT=getuigenverhalen
docker run \
    --volume ${PWD}/model-generation/notebooks:/data/notebooks \
    --volume ${PWD}/experiments/${EXPERIMENT}/corpus:/data/corpus \
    --volume ${PWD}/experiments/${EXPERIMENT}/model:/data/model \
    evidence
```
You should see two progress bars: one fro the model the other one for documents.

-  Test interactively
```shell
export EXPERIMENT=getuigenverhalen
docker run -ti --net=host\
    --volume ${PWD}/model-generation/notebooks:/data/notebooks \
    --volume ${PWD}/experiments/${EXPERIMENT}/corpus:/data/corpus \
    --volume ${PWD}/experiments/${EXPERIMENT}/model:/data/model \
   --user $(id -u):$(id -g) \
   --entrypoint /bin/bash \
    evidence
```
Note: ``--net=host`` argument may not work on Mac or Windows.

When you get the shell prompt start ``jupyter-lab``
```shell
cd /data/notebooks
jupyter-lab
```
Visit the url with the token in your browser and run the following notebooks:
1. preprocess_corpus.ipynb 
2. generate_doc2vec_model.ipynb
